### DISCUSSION TITLE
Licence Terms

### DISCUSSION THREAD

**Discussion by Oudalally** [sipsorcery-org/sipsorcery#946] [Q&A]
Hello.

Having read through the licence, I'm not entirely clear as to what I'm allowed to do with it.
I'd like to add a VOIP soft endpoint to an existing commercial product we are developing.

Am I permitted to do that using this library?

Many thanks

**Comment by sipsorcery** [comment]
Yes, you can use this library in commerical products. An acknowledgement of the source origin, for both this library and the ones it uses, is always nice. 

The only caveat is there could be some dependent libraries that use diffeent licenses such as LGPL.

**Comment by Oudalally** [comment]
That's fantastic news. 
So far, it looks like the only SIP library which will do everything we need it to cleanly!

Thank you for getting back to me so quickly, it's very much appreciated.

**Comment by cbeyersdorf** [comment]
Sorry for using this old thread, but as it is FAQ and I think this answer has changed...

Am I right that using this library inside of commercial products, is not permitted anymore, after the "License Update (May 18, 2025)"?
So for this you can only use older versions where the old licensing was used?

I'm not sure if this is intended but currently the license states commercial use isn't permitted anymore.
I think it would be nice to add this information inside this "FAQ entry".
Or maybe I'm wrong and commercial use is possible by buying an commercial license (but haven't seen something like this)?

Thanks in advance

**Comment by sipsorcery** [comment]
> Am I right that using this library inside of commercial products, is not permitted anymore, after the "License Update (May 18, 2025)"?
So for this you can only use older versions where the old licensing was used?

Nope, you can still use the software however you like, incluiding in commercial products, as per BSD-3 with one exception. The exception being:

This software must not be used to further the Apartheid policies of Israel. Use, modification, or distribution inside Israel and the Occupied Territories is strictly forbidden until the demands of the Boycott, Divestment and Sanctions movement have been met (see [full license](https://github.com/sipsorcery-org/sipsorcery?tab=License-1-ov-file) for the BDS conditions).

As the main author of the library I interpret the license conditions  to effectively mean:

- Use the software for any purpose you want, including commrecial, but if you distribute in your own product, in source or binary form, include an attribution.
- The one exception to the first point is prohibition of use in, or to the benefit of, the Apartheid State of Israel.

Hope that clears it up.

Update: I should have also said there are a handful of classes that are encumbered by GPL license conditions. If you want to use the library in a commercial application and don't want to be required to release your source code then you should remove or replace those classes.

**Comment by cbeyersdorf** [comment]
Thank you very much for the clarification.

I've read this part from the LICENSE.md:
For all other geographic regions outside of Israel and the Occupied Territories, **use**, modification, and distribution are permitted under the terms of the BSD 3-Clause "New" or "Revised" License above (Section 1), provided that any future use, modification, or distribution carries the above BDS restriction and abides by the ShareAlike and **_NonCommercial principles_.**

So I thought this means I can use it outside the restricted areas under the BSD license but not for commercial products.

So thanks again for the clarification and you're great work. :)

**Reply by sipsorcery** [reply]
> I've read this part from the LICENSE.md: For all other geographic regions outside of Israel and the Occupied Territories, **use**, modification, and distribution are permitted under the terms of the BSD 3-Clause "New" or "Revised" License above (Section 1), provided that any future use, modification, or distribution carries the above BDS restriction and abides by the ShareAlike and **_NonCommercial principles_.**
> 

You are right, that is confusing. I've removed the "and abides by the ShareAlike and **_NonCommercial principles_.**" text.
